modules/cancerBiomarkers.py

```diff
@@ -0,0 +1,352 @@
+#!/usr/bin/env python3
+"""Evidence parser for the Cancer Biomarkers database."""
+
+import argparse
+import logging
+import os
+import re
+import tempfile
+
+from pyspark.sql import DataFrame, SparkSession
+from pyspark.sql.functions import (
+    array_distinct, coalesce, col, collect_set, concat, element_at, explode, flatten, lit,
+    initcap, regexp_extract, regexp_replace, size, split, struct, translate, trim, udf, upper, when
+)
+from pyspark.sql.types import StringType, ArrayType
+
+ALTERATIONTYPE2FUNCTIONCSQ = {
+    # TODO: Map BIA
+    'MUT': 'SO_0001777',  # somatic_variant
+    'CNA': 'SO_0001563',  # copy_number_change
+    'FUS': 'SO_0001882',  # feature_fusion
+    'EXPR': None,
+    'BIA': None
+}
+
+DRUGRESPONSE2EFO = {
+    # TODO: Map Increased Toxicity and Resistance
+    'Responsive': 'GO_0042493',  # response to drug
+    'Not Responsive': 'HP_0020174',  # Refractory drug response
+    'Resistant': None,
+    'Increased Toxicity': None,
+    'Increased Toxicity (Myelosupression)': 'EFO_0007053',  # myelosuppression
+    'Increased Toxicity (Ototoxicity)': 'EFO_0006951',  # ototoxicity
+    'Increased Toxicity (Hyperbilirubinemia)': 'HP_0002904',  # Hyperbilirubinemia
+    'Increased Toxicity (Haemolytic Anemia)': 'EFO_0005558'  # hemolytic anemia
+}
+
+
+class cancerBiomarkersEvidenceGenerator():
+
+    def __init__(self):
+        # Create spark session
+        self.spark = (
+            SparkSession.builder
+            .appName('cancer-biomarkers')
+            .getOrCreate()
+        )
+
+        self.evidence = None
+
+        self.get_variantId_udf = udf(
+            cancerBiomarkersEvidenceGenerator.get_variantId,
+            StringType()
+        )
+        self.zip_alterations_with_type_udf = udf(
+            cancerBiomarkersEvidenceGenerator.zip_alterations_with_type,
+            ArrayType(ArrayType(StringType()))
+        )
+
+    def main(
+        self,
+        biomarkers_table: str,
+        source_table: str,
+        disease_table: str,
+        drug_index: str,
+        output_file: str
+    ) -> None:
+        """Loads and processes inputs to generate the Cancer Biomarkers evidence strings"""
+
+        # Import data
+        biomarkers_df = self.spark.read.csv(biomarkers_table, sep='\t', header=True)
+        source_df = self.spark.read.json(source_table).select(
+            col('label').alias('niceName'),
+            'source', 'url')
+        disease_df = self.spark.read.json(disease_table).select(
+            regexp_replace(col('name'), '_', '').alias('tumor_type'),
+            regexp_extract(col('url'), '[^/]+$', 0).alias('diseaseFromSourceMappedId'))
```

Comment:
See my comment on line 311
```suggestion
            regexp_extract(col('url'), r'[^/]+$', 0).alias('diseaseFromSourceMappedId'))
```